src/server/game/Entities/Player/Player.cpp

```diff
@@ -3136,6 +3136,8 @@ void Player::LearnSpell(uint32 spell_id, bool dependent, int32 fromSkill /*= 0*/
         SendDirectMessage(packet.Write());
     }
 
+    disabled |= DisableMgr::IsDisabledFor(DISABLE_TYPE_SPELL, spell_id, this, 0);
```

Comment:
`disabled` for known spells is a completely different thing, its a leftover from 3.3.5 when actual spell ranks were a thing you had to learn from trainers and was used to hide learned higher ranks of active talents (it also does nothing here)